src/plugin.js

```diff
@@ -362,7 +377,9 @@ const eme = function(options = {}) {
         // element between sources
         setupSessions(player);
         handleWebKitNeedKeyEvent(event, getOptions(player), player.tech_)
-          .catch(emeError);
+          .catch((error) => {
+            emeError(error, videojs.Error.WebkitKeyError);
+          });
```

Comment:
as we discussed offline:
we can analyze each rejection here, and try to use spec errors for legacy flow